src/init/power.rs

```diff
@@ -0,0 +1,122 @@
+ *                |  ███████║██║   ██║██████╔╝███████║█████╗   |              *
+ *                |  ██╔══██║██║   ██║██╔══██╗██╔══██║██╔══╝   |              *
+ *                |  ██║  ██║╚██████╔╝██║  ██║██║  ██║███████╗ |              *
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+use log::{error, info, trace};
+use std::{
+    fs::{OpenOptions, File},
+    io::Read,
+    mem,
+    path::Path,
+    slice,
+};
+
+extern crate libc;
+
+#[allow(dead_code)]
+fn syscall_reboot(action: i32) {
+    unsafe {
+        libc::reboot(action);
+    }
+}
+
+pub fn power_off() {
+    syscall_reboot(libc::LINUX_REBOOT_CMD_POWER_OFF);
+}
+
+#[allow(dead_code)]
+pub fn reboot() {
+    syscall_reboot(libc::LINUX_REBOOT_CMD_RESTART);
+}
+
+#[derive(Debug, Default, Copy, Clone)]
+#[repr(C, packed)]
+pub struct InputEvent {
+    tv_sec: u64,
+    tv_usec: u64,
+    evtype: u16,
+    code: u16,
+    value: u32,
+}
+
+// see  https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/input-event-codes.h#L191
+const KEY_POWER: u16 = 116;
+
+#[allow(dead_code)]
+pub fn spawn_power_button_listener() {
+    // TODO: detect power button devices
+    // - multiple power buttons are possible
+    // - handle reboot button with a reboot instead
+    // - devices are listed in /proc/bus/input/devices
+    let power_btn_device = Path::new("/dev/input/event0");
```

Comment:
We need constants for all of these strings unfortunately. As we move into different architecture these device paths are subject to change.

Reply:
So instead of having macos and freebsd specific functions we can just include different parts of the code which have different constant definitions. 

We have the `#DEFINE` pound define problem :(

Reply:
In general a good advice, if i think about it.
Having the constants at top level makes it so much easier to support different architectures/systems. 

I don't know how macos or freebsd handle the power button device. 
In linux, the data of a `/dev/input/eventX` can be directly read into a struct. 
```
#[repr(C, packed)]
pub struct InputEvent {
    tv_sec: u64,
    tv_usec: u64,
    evtype: u16,
    code: u16,
    value: u32,
}
```
In this particular example of power button devices, we might need to handle them differently for macos or freebsd, depending on how it works there.

Reply:
For this PR: 
* added a function parameter `power_btn_device_path: &str`
* initalization flow of pid1 uses a constant (for now). Later we want to detect the power buttons https://github.com/aurae-runtime/auraed/issues/33